src/Symfony/Component/Validator/Constraints/Xml.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Attribute\HasNamedArguments;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Exception\InvalidArgumentException;
+
+/**
+ * Validates that a value is a valid XML string.
+ *
+ * @author Mokhtar Tlili <tlili.mokhtar@gmail.com>
+ */
+#[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+class Xml extends Constraint
+{
+    public const INVALID_XML_ERROR = '0355230a-97b8-49da-b8cd-985bf3345bcf';
+
+    protected const ERROR_NAMES = [
+        self::INVALID_XML_ERROR => 'INVALID_XML_ERROR',
+    ];
+
+    #[HasNamedArguments]
+    public function __construct(
+        public string $formatMessage = 'This value is not valid XML.',
+        public string $schemaMessage = 'This value is not conform to the XML Schema.',
+        public ?string $schemaPath = null,
+        public int $schemaFlags = 0,
+        ?array $groups = null,
+        mixed $payload = null,
+    ) {
+        parent::__construct(null, $groups, $payload);
+
+        if (null !== $this->schemaPath && !file_exists($this->schemaPath)) {
```

Comment:
Let's use `is_readable()` to ensure that the file exists as well as its readability:

```suggestion
        if (null !== $this->schemaPath && !is_readable($this->schemaPath)) {
```

Reply:
In this case, you also need to change the error message: "file does not exist or cannot be read".

Reply:
Thanks for your valuable feedback. Resolved